CONTRIBUTING.md

```diff
@@ -0,0 +1,85 @@
+# Contributing
+
+`osfexport` is an open-source project. Contributions to report bugs, suggest new features or improvements, ask questions, develop the code/documentation, and any other kind of contribution are more than welcome.
+
+By contributing, you are agreeing that we may redistribute your work under [this license](https://github.com/CenterForOpenScience/osf-project-exporter?tab=Apache-2.0-1-ov-file).
+
+## Guidelines
+
+The below are guidelines about how contributions should be made, and are not necessarily hard rules.
+
+### Reporting a Bug
+For reporting bugs, create a new issue:
+- Describe what the bug is, what steps people can do to reproduce it, and what platform you were using `osfexport` on.
+- Add a `bug` label to the issue.
+
+### Suggesting a new Feature
+Create a new issue:
+- Describe what the feature is, and why it would be useful to add to the project.
+- Add a `enhancement` label to the issue.
+
+### GitHub Flow
+
+Contrbutions to code and documentation should follow the [GitHub flow model](https://docs.github.com/en/get-started/using-github/github-flow) in general. Key points are:
+- Feature branches should be branched off of develop or a release branch
+- All changes to be merged must have a Pull Request opened first.
+- Before submitting a pull request re-merge the source branch and resolve any merge conflicts
+- A branch should not be merged until it passes all checks and has been approved by one person.
+  - Checks for linting quality and if tests pass automatically runs when pull requests are made and updated.
+- Do not merge develop if you are working of a release branch and visa versa
+- Use -'s for spaces not _'s
+- Hotfixes are to be branched off main
+  - Hotfix PR should be names hotfix/brief-description
+  - A hotfix for an issue involving figshare metadata when empty lists are returned would behotfix/figshare-metadata-empty
+  - When hotfixes are merged a new branch will be created bumping the minor version ie hotfix/0.1.3 and the other PR will be merged into it
+
+The naming convention for branches is: <issue-number>-<brief-issue-description>. For example, a branch for issue 5 `Make tests run faster` would be named `5-make-tests-run-faster`.
+
+### New Releases
+
+To make a new release:
+- Make a Pull Request from development to main with a commit to bump the version number in `pyproject.toml` to a new version.
+- The PR should not be merged until all tests pass, one human has reviewed the PR, and the new version can be built as a Python package.
+  - The package should be able to be built using:
+    ```
+    python3 -m pip install --upgrade build
+    python -m build
+    ```
+- Once all tests pass, the PR has been reviewed, and the package can be built, merge into main
+- Go to Releases and create a new release for the new version.
```

Comment:
Should we leave out htis bit about making a release? I feel tlike this is something the COS maintainers should do, not other contributors.